Fix miri imports for inclusion in libstd

Aaron1011 opened the pull request.

RalfJung commented:
This blocks fixing the Miri component, would be good to get it landed ASAP. :)

alexcrichton commented on src/symbolize/miri.rs:
```diff
@@ -1,9 +1,18 @@
+    pub use crate::*;
+}
+#[cfg(not(backtrace_in_libstd))]
+extern crate std as mystd;
```
Instead of doing this, can the `filename` method be behind `#[cfg(feature = "std")]`? I don't think that the standard library uses that method.

alexcrichton commented:
If you'd like to ensure that this continues to work over time you can add another thing to test [here](https://github.com/rust-lang/backtrace-rs/blob/master/.github/workflows/main.yml#L85)

Aaron1011 commented:
@alexcrichton: There's some additional work needed to get tests passing under Miri - I plan to open a follow-up PR.

alexcrichton commented:
Oh it's also fine to just test a build, no need to run tests just yet if they don't work.

Aaron1011 commented:
@alexcrichton: I've addressed your comment. If it's okay with you, I'd like to do any CI changes in a follow-up PR, so that we get Miri building again quickly.

alexcrichton commented:
👍

Aaron1011 commented:
@alexcrichton: Do we need to make a new release before bumping the submodule in rustc?

alexcrichton commented:
I believe it's just a submodule update, so no a release isn't needed.
